linux_os/guide/services/ssh/ssh_private_keys_have_passcode/rule.yml

```diff
@@ -0,0 +1,38 @@
+documentation_complete: true
+
+prodtype: ol8
+
+title: 'OpenSSH Service Must Use Passcode for Their Private Keys'
+
+description: |-
+    Verify the SSH private key files have a passcode.
+    For each private key stored on the system, use the following command:
+
+    <pre>$ sudo ssh-keygen -y -f /path/to/file</pre>
+
+    If the contents of the key are displayed, without asking a passphrase this is a finding.
+
+rationale: |-
+    If an unauthorized user obtains access to a private key without a passcode, that user would
+    have unauthorized access to any system where the associated public key has been installed.
+
+severity: medium
+
+references:
+    disa: CCI-000186
+    nist: IA-5(2),IA-5(2).1
+    srg: SRG-OS-000067-GPOS-00035
+    stigid@ol8: OL08-00-010100
+
+ocil_clause: Any contents were displayed without asking a passphrase
+
+ocil: |-
+    For each private key stored on the system, use the following command:
+
+    <pre>$ sudo ssh-keygen -y -f /path/to/file</pre>
+
+fixtext: |-
+    Replace existing private keys with new ones protected by a passphrase. Create a new private
```

Comment:
It it valid to simply set a password on existing key without a password? Unless there are any reason to consider the key was already leaked, sounds reasonable to include a password on existing keys instead of generating new keys. What do you think?

Reply:
Makes sense